flink-runtime/src/main/java/org/apache/flink/runtime/clusterframework/TaskExecutorResourceUtils.java

```diff
@@ -103,6 +105,12 @@ public static ResourceProfile generateDefaultSlotResourceProfile(TaskExecutorRes
 			.build();
 	}
 
+	public static List<ResourceProfile> createWorkerSlotProfiles(TaskExecutorResourceSpec taskExecutorResourceSpec) {
```

Comment:
code style nit: better to put this function before `generateDefaultSlotResourceProfile`
we usually put functions used in other functions after them.

Reply:
I would also call this method `createDefaultWorkerSlotProfiles`